uninstall_zabbix.rb

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env ruby
+#
+# uninstall_zabbix.rb
+#
+# This script coordinates the uninstall of zabbix from the head nodes.
+#
+# Provisos:
+#
+# To use from the command line:
+#
+#  1. If necessary, configure your local rubygems mirror.
+#     Replace 'http://mirror.example.com' with your actual mirror.
+#     ```
+#     bundle config mirror.https://rubygems.org http://mirror.example.com
+#     ```
+#
+#  2. Run 'bundle install --deployment' on a bootstrap node with
+#     access to a rubygems mirror.
+#
+#  3. If not already using the target bootstrap, sync the updated
+#     repository, including 'vendor' directory, to the target
+#     bootstrap.
+#
+#  4. Run 'bundle exec ./uninstall_zabbix.rb -p <mysql root password>
+#      ' to begin the process.
+#
+# It is also possible to use methods from this script at a ruby REPL
+# instead of running the script from a UNIX shell.  To load methods
+# into `irb`:
+#
+#  1. Change to the repo directory.
+#
+#  2. Verify that dependencies are installed:
+#     `bundle list`
+#
+#  3. Run irb inside the repo directory.
+#     `bundle exec irb`
+#
+#  4. Load this file.
+#     `irb(main):001:0> load ./uninstall_zabbix.rb`
+#
+
+require 'chef/provisioning/transport/ssh'
+require 'mixlib/shellout'
+require 'pry'
+require 'timeout'
+require 'optparse'
+require 'json'
+require 'rubygems'
+require 'ohai'
+require 'mixlib/shellout'
+
+def get_entry(name)
+  parse_cluster_txt.select { |e| e['runlist'].include? name }.first
+end
+
+def head_nodes
+  parse_cluster_txt.select { |e| e['runlist'].include? 'Head' }
+end
+
+def worker_nodes
+  parse_cluster_txt.select { |e| e['runlist'].include? 'Worker' }
+end
+
+def virtualbox_vm?(entry)
+  /^08:00:27/.match(entry['mac_address'])
+end
+
+def parse_cluster_txt
+  fields =
+    %w(hostname mac_address ip_address ilo_address cobbler_profile
+       dns_domain runlist)
+  # This is really gross because Ruby 1.9 lacks Array#to_h.
+  File.readlines(File.join(repo_dir, 'cluster.txt'))
+    .map { |line| Hash[*fields.zip(line.split(' ')).flatten(1)] }
+end
+
+def repo_dir
+  File.dirname(__FILE__)
+end
+
+def check_status(c, on_fail_msg, on_success_msg, dofail = false)
```

Comment:
Hi Esther, could you please provide a function comment here? (What is `c`?)

E.g.
````
##################
# check_status -- check status via parsing a c object
# expects: c - a c object
#              on_fail_msg - string
#              on_success_msg - string
#              dofail - to fail for fun and profit without checking something
# side affect: prints a message directly to the user via puts()